x-pack/plugins/security_solution/cypress/integration/detection_alerts/attach_to_case.spec.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { newRule } from '../../objects/rule';
+import { ROLES } from '../../../common/test';
+
+import { waitForAlertsIndexToBeCreated, waitForAlertsPanelToBeLoaded } from '../../tasks/alerts';
+import { createCustomRuleActivated } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import { waitForAlertsToPopulate } from '../../tasks/create_new_rule';
+import { login, loginAndWaitForPage, waitForPageWithoutDateRange } from '../../tasks/login';
+import { refreshPage } from '../../tasks/security_header';
+
+import { DETECTIONS_URL } from '../../urls/navigation';
+import { ATTACH_ALERT_TO_CASE_BUTTON } from '../../screens/alerts_detection_rules';
+
+describe('Alerts timeline', () => {
+  before(() => {
+    // First we login as a privileged user to create alerts.
+    cleanKibana();
+    loginAndWaitForPage(DETECTIONS_URL, ROLES.platform_engineer);
+    waitForAlertsPanelToBeLoaded();
+    waitForAlertsIndexToBeCreated();
+    createCustomRuleActivated(newRule);
+    refreshPage();
+    waitForAlertsToPopulate();
+
+    // Then we login as read-only user to test.
+    login(ROLES.reader);
+  });
+
+  beforeEach(() => {
+    waitForPageWithoutDateRange(DETECTIONS_URL, ROLES.reader);
+    waitForAlertsToPopulate();
+  });
+
+  it('should not allow a read only user to attach alerts to cases', () => {
+    cy.get(ATTACH_ALERT_TO_CASE_BUTTON).first().should('be.disabled');
+  });
+});
```

Comment:
This looks fine so far. The suggestion with permissions for e2e tests we have been making to each other is that we have 1 test which shows it being disabled for one role and then a second test which shows it being enabled for 1 more role.

This way if someone breaks the code so that it is disabled for all roles we can catch it.